framework/log/DbTarget.php

```diff
@@ -60,6 +60,12 @@ public function init()
      */
     public function export()
     {
+        if ($this->db->getTransaction()) {
+            $this->db = clone $this->db;
+            $this->db->pdo = null;
```

Comment:
sorry for the late reply, but does this work in a master-slave setup? The situation is a bit more complex there: https://github.com/yiisoft/yii2/blob/1c07c429cf54e58ccb494f05ed889e6dce284bbe/framework/db/Connection.php#L578-L597

also transaction property of the DB is not reset so for each call to export another connection is opened even though there is no transaction on current connection anymore.